Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -46,6 +47,8 @@ @implementation SentrySessionReplayIntegration {
     SentryReplayOptions *_replayOptions;
     SentryNSNotificationCenterWrapper *_notificationCenter;
     SentryOnDemandReplay *_resumeReplayMaker;
+    id<SentryRateLimits> _rateLimits;
+    BOOL _rateLimited;
```

Comment:
```suggestion
   // We need to use this variable to identify whether rate limiting was ever activated for session replay in this session, instead of always looking for the rate status in `SentryRateLimits`
   // This is the easiest way to ensure segment 0 will always reach the server, because session replay absolutely needs segment 0 to make replay work.
    BOOL _rateLimited;
```